### PR TITLE
sealos-image-hub

### PR DESCRIPTION
**Pull request by zjy365**

### CODE REVIEW

File: frontend/providers/imagehub/Makefile [hunk 1]
```diff
@@ -0,0 +1,48 @@
+
+SERVICE_NAME=zhujingyang/sealos-image-hub
```
Comment (LeezQ) [inline frontend/providers/imagehub/Makefile:2]: `zhujingyang/sealos-image-hub`  Is the image url here incorrect? It should be placed in the public repository.

File: frontend/providers/imagehub/deploy/manifests/frontend.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      containers:
+        - name: desktop-frontend
+          securityContext:
+            runAsNonRoot: true
+            runAsUser: 1001
+            allowPrivilegeEscalation: false
+            capabilities:
+              drop:
+                - "ALL"
+          image: zhujingyang/desktop-frontend:latest
```
Comment (LeezQ) [inline frontend/providers/imagehub/deploy/manifests/frontend.yaml:87]: Here is the same

**Review by LeezQ — APPROVED** [review]

File: frontend/providers/imagehub/Makefile (hunk 1 above)
Comment (zjy365) [reply frontend/providers/imagehub/Makefile:2]: It will become a public warehouse later

File: frontend/providers/imagehub/src/layout/index.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    return createSealosApp({
+      appKey: 'sealos-image-hub'
+    });
+  }, []);
+
+  useEffect(() => {
+    const initApp = async () => {
+      try {
+        const result = await sealosApp.getUserInfo();
+        setSession(result);
```
Comment (zzjin) [inline frontend/providers/imagehub/src/layout/index.tsx:22]: When deploying apps along side with custom unique url domain, we do not want to store sessions under local storage for safety purpose.
Maybe getSession will just save user infos as document variables is enough.

File: frontend/providers/imagehub/src/stores/session.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  setSessionProp: (key: keyof Session, value: any) => void;
+  getSession: () => Session;
+  delSession: () => void;
+  isUserLogin: () => boolean;
+  getKubeconfigToken: () => string;
+};
+
+const useSessionStore = create<SessionState>()(
+  devtools(
+    persist(
```
Comment (zzjin) [inline frontend/providers/imagehub/src/stores/session.ts:20]: Just remove this `persist` wrapper is enough./

**Review by zzjin — APPROVED** [review]

**Pull request merged**